### PR TITLE
chore: (multi-domain) update cross-origin error

### LINKED ISSUE #20974: [Multi-origin]: Improve error message when attempting to visit different origins at same level
### What would you like?

Improve the [error message](https://github.com/cypress-io/cypress/blob/8a78b5c127a937df58d96e4b1d1f689416ce387d/packages/driver/src/cypress/error_messages.ts#L2093-L2115) when a user attempts to visit 2 different origins at the same level in a test. 

It should mention that cy.origin can be used to accomplish that.

Examples:

```js
cy.visit('domain1.com')
cy.visit('domain2.com')
// ... more commands on domain2.com ...
```

should instead be:

```js
cy.visit('domain1.com')
cy.origin('domain2.com', () => {
  cy.visit('domain2.com')
  // ... more commands on domain2.com ...
})
```

And this applies within cy.origin as well:

```js
cy.visit('domain1.com')
cy.origin('domain2.com', () => {
  cy.visit('domain2.com')
  cy.visit('domain3.com')
 // ... more commands on domain3.com ...
})
```

should instead be:

```js
cy.visit('domain1.com')
cy.origin('domain2.com', () => {
  cy.visit('domain2.com')
})
cy.origin('domain3.com', () => {
  cy.visit('domain3.com')
 // ... more commands on domain3.com ...
})
```

In the future, we could also recommended nested cy.origins for the second example once they are supported.

### Why is this needed?

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by mschile**
Closes #20974

### User facing changelog

n/a

### Additional details

* Updated the cross-origin error message to inform the user to use `cy.origin`

### How has the user experience changed?

Experimental flag off:
Screen Shot 2022-04-08 at 8 42 07 AM

Experimental flag on:
Screen Shot 2022-04-08 at 8 42 57 AM

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/src/cypress/error_messages.ts [hunk 1]
```diff
@@ -1133,6 +1131,107 @@ export default {
     no_global: `Angular global (\`window.angular\`) was not found in your window. You cannot use ${cmd('ng')} methods without angular.`,
   },
 
+  origin: {
```
Comment (mschile) [inline packages/driver/src/cypress/error_messages.ts:1134]: No changes in this section, just moved the `origin` section to it's correct alphabetical location.

File: packages/driver/src/cypress/error_messages.ts [hunk 2]
```diff
@@ -1000,12 +1000,10 @@ export default {
...
 
         A command that triggers cross origin navigation must be immediately followed by a ${cmd('origin')} command:
-
-        \`\`\`
-        cy.origin('${crossOriginUrl.originPolicy}', () => {
-          <commands targeting ${crossOriginUrl.origin} go here>
-        })
-        \`\`\`
+        
+        \`cy.origin('${crossOriginUrl.originPolicy}', () => {\`
```
Comment (mjhenkes) [inline packages/driver/src/cypress/error_messages.ts:1004]: nice work around

File: packages/driver/src/cypress/error_messages.ts [hunk 3]
```diff
@@ -2103,13 +2117,11 @@ export default {
...
-            > '{{previousUrl}}'
+            > '${args.previousUrl.origin}'
 
           You're attempting to visit this URL:
 
-            > '{{attemptedUrl}}'
-
-          You may need to restructure some of your test code to avoid this problem.`,
+            > '${args.attemptedUrl.origin}'`,
         docsUrl: 'https://on.cypress.io/cannot-visit-different-origin-domain',
```
Comment (mjhenkes) [inline packages/driver/src/cypress/error_messages.ts:2125]: should we flex this docs url depending on if they have the feature flag enabled?, Send them to `cy.origin` if it is enabled?

Comment (mschile) [reply packages/driver/src/cypress/error_messages.ts:2125]: I was hoping this doc would be updated to inform the user why they received this error and link to `cy.origin`.

Comment (mjhenkes) [reply packages/driver/src/cypress/error_messages.ts:2125]: ok, this is fine for now, lets double check with DX next week.

**Review by mjhenkes — APPROVED** [review]

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**